### PR TITLE
Translations for Email Notifications

### LINKED ISSUE #547: Translated Emails
Previously this was quite hard to do because the email templates where precompiled html files but now that we are using React as a templating engine for emails it should be possible to use `react-i18next` to translate these emails as well.

The subject lines will need a separate solution (hopefully using `i18next`) since they are not part of the templates.

<sub>[RAL-247](https://linear.app/rallly/issue/RAL-247/translated-emails)</sub>

## Upvote & Fund

- We're using [Polar.sh](https://polar.sh/lukevella) so you can upvote and help fund this issue.
- We receive the funding once the issue is completed & confirmed by you.
- Thank you in advance for helping prioritize & fund our backlog.

<a href="https://polar.sh/lukevella/rallly/issues/547">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/lukevella/rallly/issues/547/pledge.svg?darkmode=1">
  Fund with Polar
</picture>
</a>

### PR DESCRIPTION
**Pull request by NikoHadouken**
## Description

**This PR adds translations for transactional email notifications sent by rallly.**

**fixes #547**

## Checklist

Please check off all the following items with an "x" in the boxes before requesting a review.

- [x] I have performed a self-review of my code
- [x] My code follows the code style of this project
- [ ] I have commented my code, particularly in hard-to-understand areas

## Todo
- [x] support translations in email content
- [x] support translations in email subjects
- [ ] add translations support for every email template
- [ ] add translations for all supported languages

## Summary by CodeRabbit

- **New Features**
	- Enhanced email templates with dynamic subject lines for various email notifications, improving contextual relevance.
	- Added localization support for German and English email templates, enhancing user experience through personalized content.
	- Introduced a method for generating email subjects based on provided properties, allowing for tailored communications.

- **Bug Fixes**
	- Removed dynamic tokens from email subjects to enhance security and reduce potential confusion for users.

- **Chores**
	- Updated internationalization functionality to support multiple languages seamlessly.

### CODE REVIEW

**Comment by NikoHadouken** [comment]
It would probably better to refactor some `i18next` related code into the `@rallly/languages` package.
* apps/web/src/app/i18n.ts
* apps/web/src/app/i18n/settings.ts
* packages/emails/src/i18n.ts

We could put the instance creation function and shared settings in there.

**Comment by NikoHadouken** [comment]
Is `packages/emails/locales` a good place for notification translation files or should we try to keep all translation files in one place?

**Comment by lukevella** [comment]
Hey @NikoHadouken , I think you're on the right track putting the translation files in `packages/emails/locales`. We shouldn't mix them with the web translations because we don't want these loaded on in the client application and while server rendering Next.js pages. They're only needed on specific routes - just the trpc endpoint at the moment. At the same time, we don't want to be loading the entire web app translation files when all we want is to translate the emails so having separate instances is the right way to go.

Trying to consolidate everything into `@rallly/languages` could work but lets focus on getting things working as is. I think you've done a great job organizing things so far. Well done!

File: packages/emails/src/templates/new-comment.tsx [hunk 1]
```diff
@@ -1,8 +1,10 @@
-import { defaultEmailContext } from "./_components/email-context";
+import { defaultEmailContext, EmailContext } from "./_components/email-context";
 import NotificationEmail, {
   NotificationBaseProps,
 } from "./_components/notification-email";
 import { Text } from "./_components/styled-components";
+import { Trans } from "next-i18next";
```
Comment (lukevella) [inline packages/emails/src/templates/new-comment.tsx:6]:
```suggestion
import { Trans } from "react-i18next/TransWithoutContext";
```

**Comment by NikoHadouken** [comment]
@lukevella Good dicision. I will update the `i18next` part in the emails package to behave like the one in `web` and implement translations in the other templates.

**Comment by lukevella** [comment]
I'm also, considering going through the templates and updating their content so no need to go through and update all of them. If it's working for one email I can go through and update the rest. It will be easier to source translations through Crowdin as well.

**Pull request merged**